### DISCUSSION TITLE
Proper way to handle client-only elements like HTML Canvas?

### DISCUSSION THREAD

**Discussion by ffxsam** [nuxt/nuxt#17311] [Questions] [answered]
Example here:

https://stackblitz.com/edit/nuxt-starter-z3iba3?file=app.vue,src%2Fcomponents%2FCanvasThing.vue,src%2Fwaveform-progress.ts

Look at app.vue, then CanvasThing.vue, then waveform-progress.ts. Is there a cleaner way to handle this? Nuxt won't build this app due to a reference to `canvas`.

I suppose the thing that confuses me is: if I put a `console.log()` call in CanvasThing.vue, which is wrapped by `client-only`, I expectedly don't see that log in the shell - only in the browser. If that's the case, why can't I import client-only libraries like `canvas` from within CanvasThing.vue?

**Comment by danielroe** [comment, marked as answer]
See https://github.com/nuxt/framework/pull/5750, which should both explain the issue and hopefully resolve it for you when merged.

**Reply by ffxsam** [reply]
Thanks, Daniel! Also (go figure) if I had followed the Nuxt folder structure properly and not made my own `src` subfolder to put components in, then the `Component.client.vue` trick would've saved me, looks like (though I didn't see the `.client.vue` trick mentioned in the docs).

To anyone following this thread: I updated the StackBlitz example so it's in a working state.
